Documentation/components-versioning.md

```diff
@@ -0,0 +1,37 @@
+Component versioning
+====================
+
+We are using [Semantic Versioning](https://semver.org/) in our components, and
+we are following more or less exactly the definition of Major, Minor and Patch
+versions defined there. However, since Semantic Versioning was primarily written
+for libraries, there are some key differences in what we call an API in Mender's
+context.
+
+### Count towards new major version
+
+* Removing or changing command line options
+
+* Removing or changing existing REST API
+
+* Edge case: Changing default behavior, with the old behavior still
+  available. An example is changing the default artifact format version in
+  `mender-artifact`. We have opted for upgrading the major version in this case,
+  but this could also go into a minor release
+
+### Count towards new minor version
+
+* Adding command line options
+
+* Adding new REST API, or adding fields to responses of existing REST API
+
+* Doing a database migration that is incompatible with the old schema (downgrade
+  usually not possible without restoring a backup)
+
+### Count towards new patch version
+
+* Any change to our Golang API. For example, the `mender-artifact` library has
```

Comment:
Say "internal API" instead of "Golang API"?

Reply:
I think that's less clear. Which API is the "internal API" exactly? Does it include REST? I think it's better to say Golang API, because it's very specific. We don't have any public Golang API ATM as far as I know, and I don't think we're planning any either.